routes/api/v1.js

```diff
@@ -424,6 +427,120 @@ api.declare({
+    "which type of credentials is returned. Please note that the `level`",
+    "parameter is required in the scope guarding access.",
+    "",
+    "The credentials are set of expire after an hour, but this behavior may be",
+    "subject to change. Hence, you should always read the `expires` property",
+    "from the response, if you intent to maintain active credentials in your",
+    "application.",
+    "",
+    "Please notice that your `prefix` shouldn't start with slash `/`, it is",
+    "allowed for completeness, but it will result in two slashes both in the",
+    "scope and in the bucket. Therefore this pattern is strongly discouraged.",
+    "Also note that if your `prefix` doesn't end in a slash `/` you will not",
+    "be required to insert one. This is mainly a concern when assigning scopes",
+    "to users and doing this right will prevent poor behavior."
+  ].join('\n')
+}, function(req, res) {
+  var level   = req.params.level;
+  var bucket  = req.params.bucket;
+  var prefix  = req.params.prefix;
+
+  // Validate that a proper value was given for level
+  if (level !== 'read-write' && level !== 'read-only') {
+    return res.status(400).json({
+      message:      "the 'level' URL parameter must be read-only or read-write",
+      levelGiven:   level
+    });
+  }
+
+  // Check that the client is authorized to access given bucket and prefix
+  if (!req.satisfies({
+    level:      level,
+    bucket:     bucket,
+    prefix:     prefix
+  })) {
+    return;
+  }
+
+  return this.sts.getFederationToken({
+    Name:               'TemporaryS3ReadWriteCredentials',
+    Policy:             JSON.stringify({
+      Version:          '2012-10-17',
+      Statement:[
+        {
+          Sid:            'ReadWriteObjectsUnderPrefix',
+          Effect:         'Allow',
+          Action:         (level === 'read-write' ? [
+            's3:GetObject',
+            's3:PutObject',
+            's3:DeleteObject'
+          ] : [
+            's3:GetObject'
+          ]),
+          Resource: [
+            'arn:aws:s3:::{{bucket}}/{{prefix}}*'
+              .replace('{{bucket}}', bucket)
+              .replace('{{prefix}}', prefix)
+          ]
+        }, {
+          Sid:            'ListObjectsUnderPrefix',
+          Effect:         'Allow',
+          Action: [
+            's3:ListBucket'
+          ],
+          Resource: [
+            'arn:aws:s3:::{{bucket}}'
+              .replace('{{bucket}}', bucket)
+          ],
+          Condition: {
+            StringLike: {
+              's3:prefix': [
+                '{{prefix}}*'.replace('{{prefix}}', prefix)
+              ]
+            }
+          }
+        }, {
+          Sid:            'GetBucketLocation',
+          Effect:         'Allow',
+          Action: [
+            's3:GetBucketLocation'
+          ],
+          Resource: [
+            'arn:aws:s3:::{{bucket}}'
+              .replace('{{bucket}}', bucket)
+          ]
+        }
+      ]
+    }),
```

Comment:
Reasonably sure... :)

I'm not scared of all the `get` type actions. 

The only thing that is slightly _interesting_ is the prefix thing...
But all of this comes from: http://blogs.aws.amazon.com/security/post/Tx1P2T3LFXXCNB5/Writing-IAM-policies-Grant-access-to-user-specific-folders-in-an-Amazon-S3-bucke

Note, the tests check for both access and access denied. So it works :)